server/services/recommendedVersionsService.ts

```diff
@@ -0,0 +1,219 @@
+    ) {
+      result.metadata.source = result.metadata.source === 'none' ? 'none' : 'partial'
+    }
+
+    this.cache = {
+      versions: result,
+      expiresAt: now + this.ttlMillis,
+    }
+
+    logger.debug(
+      `[RecommendedVersions] Source=${result.metadata.source}, helm(gpa=${
+        result.helm_dependencies.generic_prometheus_alerts || 'missing'
+      }, gs=${result.helm_dependencies.generic_service || 'missing'}), gradle(hgsb=${
+        result.gradle.hmpps_gradle_spring_boot || 'missing'
+      })`,
+    )
+
+    return result
+  }
+
+  private parseVersionValue(raw: unknown): string | undefined {
+    if (raw === null || raw === undefined) return undefined
+    if (typeof raw === 'string' || typeof raw === 'number') return String(raw)
+    if (typeof raw === 'object') {
+      const obj = raw as Record<string, unknown>
+      return (obj.ref as string) || (obj.version as string) || undefined
+    }
+    return undefined
+  }
+
+  private async fetchRecommendedVersionsFromStrapi(): Promise<{
+    helm_generic_prometheus_alerts?: string
+    helm_generic_service?: string
+    gradle_hmpps_gradle_spring_boot?: string
+  } | null> {
+    try {
+      const templateComponentName = config.recommendedVersions.componentName
+      if (!templateComponentName) {
+        logger.warn('[RecommendedVersions] Missing template component name in config.recommendedVersions.componentName')
+        return null
+      }
+
+      const componentNameVariants = Array.from(
+        new Set(
+          [
+            templateComponentName,
+            templateComponentName.replace(/_/g, '-'),
+            templateComponentName.replace(/-/g, '_'),
+          ].filter(Boolean),
+        ),
+      )
+
+      let component: unknown | null = null
+      for (const variantName of componentNameVariants) {
+        try {
+          // eslint-disable-next-line no-await-in-loop -- sequential attempts are intentional to avoid unnecessary parallel requests
+          component = await this.serviceCatalogueService.getComponent({ componentName: variantName })
+          logger.info(`[RecommendedVersions] Strapi component match: name=${variantName}`)
+          break
+        } catch (e) {
+          logger.warn(`[RecommendedVersions] Strapi lookup failed for name=${variantName}: ${String(e)}`)
+        }
+      }
+
+      if (!component) {
+        logger.warn(
+          `[RecommendedVersions] Strapi component not found. Tried variants=${JSON.stringify(componentNameVariants)}`,
+        )
+        return null
+      }
+
+      const versions = (component as unknown as { versions?: Record<string, unknown> }).versions || {}
+      const values = (component as unknown as { values?: Record<string, unknown> }).values || {}
+
+      // Preferred keys from versions
+      const helmDeps = (versions.helm_dependencies as Record<string, unknown>) || {}
+      let helmGenericPrometheusAlerts = this.parseVersionValue(helmDeps?.generic_prometheus_alerts)
```

Comment:
Feel free to tell me I'm wrong, but rather than all these possible calls to `parseVersionValue`, could you not just set all the attributes like `helmGenericPrometheusAlerts` to the raw value, then once you've gone through all of your checks, then just reset them via this at the end, so your return would be:

```
return {
          helm_generic_prometheus_alerts: this.parseVersionValue(helmGenericPrometheusAlerts),
          helm_generic_service: this.parseVersionValue(helmGenericService),
          gradle_hmpps_gradle_spring_boot: this.parseVersionValue(gradleHmppsGradleSpringBoot),
        }
```

Maybe?